### PR TITLE
fixes #370

### LINKED ISSUE #370: duplicated layers not allowed
Actually the identifier of a layer in each action and tool is the layer name. This causes problems if we have duplicated layers (e.g. with different styles). 

We should allow to pass an `id` property from the configuration, by default initialized with the layer name (@source if possible).
Then all the actions, reducers and internal calls to should use the id to identify the layer

### PR DESCRIPTION
**Pull request by kappu72**

### CODE REVIEW

File: web/client/examples/layertree/app.jsx [hunk 1]
```diff
@@ -55,16 +56,18 @@ const getGroup = (group, allLayers) => {
         }, []);
 
     return assign({}, {
+        id: groupName,
```
Comment (offtherailz) [inline web/client/examples/layertree/app.jsx]: Clarify why also groups need an id

File: web/client/utils/ConfigUtils.js [hunk 2]
```diff
@@ -72,7 +72,7 @@ var ConfigUtils = {
         other.center = ConfigUtils.getCenter(other.center);
         return {
             map: other,
-            layers: layers.map(ConfigUtils.setBingKey),
+            layers: layers.map(ConfigUtils.setBingKey).map(ConfigUtils.setLayerId),
```
Comment (offtherailz) [inline web/client/utils/ConfigUtils.js]: So the id is mandatory and this function is an helper to assign the name if the id is not present.

Comment (kappu72) [inline web/client/utils/ConfigUtils.js]: Yes It'a mandatory and setLayerId is an helper function that set id equal to layer's name if id is missing

File: web/client/examples/layertree/app.jsx (hunk 1 above)
Comment (kappu72) [inline web/client/examples/layertree/app.jsx]: In layers reducer deepChange function is used both for layers and group, now it checks for a generic node id not name

File: web/client/utils/ConfigUtils.js [hunk 3]
```diff
@@ -268,6 +268,12 @@ var ConfigUtils = {
             layer.apiKey = defaultConfig.bingApiKey;
         }
         return layer;
+    },
+    setLayerId: function(layer) {
+        if (!layer.id) {
+            layer.id = layer.name;
```
Comment (offtherailz) [inline web/client/utils/ConfigUtils.js]: We should use a progressive number to set unique id anyway, for duplicates too.

**Pull request merged**